test/binder/binder_test.cpp

```diff
@@ -152,7 +163,10 @@ TEST_F(BinderCorrectnessTest, SelectStatementTest) {
   }
 
   // Test select from different table instances from the same physical schema
-  binder.reset(new binder::BindNodeVisitor());
+  delete binder->txn;
```

Comment:
@pmenon is right. This is not how you do this. It should be in the deconstructor.